app/src/main/java/ca/rmen/android/poetassistant/settings/SettingsActivity.java

```diff
@@ -89,6 +96,24 @@ public void onCreatePreferences(Bundle bundle, String s) {
             if (voicePreference.getEntries() == null || voicePreference.getEntries().length < 2) {
                 getPreferenceScreen().removePreference(voicePreference);
             }
+            Preference systemTtsSettings = findPreference(Settings.PREF_SYSTEM_TTS_SETTINGS);
+            Intent intent = systemTtsSettings.getIntent();
+            if (intent.resolveActivity(getActivity().getPackageManager()) == null) {
+                getPreferenceScreen().removePreference(systemTtsSettings);
+            }
+        }
+
+        @Override
+        public void onResume() {
+            super.onResume();
+            EventBus.getDefault().register(this);
+            Tts.getInstance(getContext()).restart();
```

Comment:
We only really need to restart TTS if the user had opened the system tts settings, and we're just now coming back from it. But, it's simpler code to restart it each time in `onResume()`.  Otherwise we'd have to have an `OnPreferenceClickListener` for the system tts settings pref, which sets some flag saying "the user opened the system settings", and in `onResume` we'd have to check that flag and only restart tts if it's true.  But if we do that, we have to take into account configuration changes (device rotation).  Too much code...